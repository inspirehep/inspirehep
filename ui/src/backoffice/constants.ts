import { List, Map } from 'immutable';

import { AUTHORS_PID_TYPE, LITERATURE_PID_TYPE } from '../common/constants';

export enum WorkflowTypes {
  AUTHOR_CREATE = 'AUTHOR_CREATE',
  AUTHOR_UPDATE = 'AUTHOR_UPDATE',
  HEP_CREATE = 'HEP_CREATE',
}

export const WORKFLOW_TYPES = {
  [WorkflowTypes.AUTHOR_CREATE]: AUTHORS_PID_TYPE,
  [WorkflowTypes.AUTHOR_UPDATE]: AUTHORS_PID_TYPE,
  [WorkflowTypes.HEP_CREATE]: LITERATURE_PID_TYPE,
};

export interface WorkflowStatus {
  key: string;
  doc_count: number;
}

export interface WorkflowType {
  key: string;
  doc_count: number;
}

export interface Collection {
  key: string;
  value: string | undefined;
}

export interface WorkflowCardProps {
  type: Map<string, any>;
  statuses: List<Map<string, any>>;
}

export type WorkflowTypeKey = 'AUTHOR_CREATE' | 'AUTHOR_UPDATE' | 'HEP_CREATE';
