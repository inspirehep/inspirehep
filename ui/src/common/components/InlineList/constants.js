export const DEFAULT_SEPARATE_ITEMS_CLASS = 'separate-items-with-comma';
export const SEPARATE_ITEMS_CLASSNAMES = [
  'separate-items-with-semicolon',
  'separate-items-with-comma',
  'separate-items-with-and',
  'separate-items-with-middledot',
];
