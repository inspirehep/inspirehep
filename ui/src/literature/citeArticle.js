import http from '../common/http';

export default async function(format, recordId) {
  const response = await http.get(`/literature/${recordId}`, {
    headers: {
      Accept: `application/${format}`,
    },
  });
  return response.data;
}
