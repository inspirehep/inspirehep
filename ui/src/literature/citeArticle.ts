import http from '../common/http.ts';

export default async function(format, recordId) {
  const response = await http.get(`/literature/${recordId}`, {
    headers: {
      Accept: format,
    },
  });
  return response.data;
}
