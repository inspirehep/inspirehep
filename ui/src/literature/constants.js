export const CITE_FORMAT_OPTIONS = [
  { value: 'x-bibtex', display: 'BibTeX' },
  { value: 'vnd+inspire.latex.eu+x-latex', display: 'LaTeX (EU)' },
  { value: 'vnd+inspire.latex.us+x-latex', display: 'LaTeX (US)' },
];
export const CITE_FORMAT_VALUES = CITE_FORMAT_OPTIONS.map(
  option => option.value
);

export const MAX_CITEABLE_RECORDS = 1000;
