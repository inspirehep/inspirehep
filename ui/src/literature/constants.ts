const BIBTEX = 'application/x-bibtex';
const LATEX_EU = 'application/vnd+inspire.latex.eu+x-latex';
const LATEX_US = 'application/vnd+inspire.latex.us+x-latex';
export const CV = 'text/vnd+inspire.html+html';

export const CITE_FORMAT_OPTIONS = [
  { value: BIBTEX, display: 'BibTeX' },
  { value: LATEX_EU, display: 'LaTeX (EU)' },
  { value: LATEX_US, display: 'LaTeX (US)' },
  { value: CV, display: 'CV (html)' },
];

export const CITE_FILE_FORMAT = {
  [BIBTEX]: { mimetype: 'application/x-bibtex', extension: 'bib' },
  [LATEX_EU]: { mimetype: 'application/x-latex', extension: 'tex' },
  [LATEX_US]: { mimetype: 'application/x-latex', extension: 'tex' },
  [CV]: { mimetype: 'text/html', extension: 'html' },
};

export const CITE_FORMAT_VALUES = CITE_FORMAT_OPTIONS.map(
  (option) => option.value
);

export const MAX_CITEABLE_RECORDS = 1000;

export const MAX_BULK_ASSIGN = 250;
