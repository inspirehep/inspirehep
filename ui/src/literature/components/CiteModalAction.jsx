import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Modal, Button, Row, Icon, Alert } from 'antd';
import { CopyToClipboard } from 'react-copy-to-clipboard';

import SelectBox from '../../common/components/SelectBox';
import ListItemAction from '../../common/components/ListItemAction';
import IconText from '../../common/components/IconText';
import EventTracker from '../../common/components/EventTracker';
import { CITE_FORMAT_OPTIONS, CITE_FORMAT_VALUES } from '../constants';
import citeArticle from '../citeArticle';
import { downloadTextAsFile } from '../../common/utils';

class CiteModalAction extends Component {
  constructor(props) {
    super(props);
    this.onCiteClick = this.onCiteClick.bind(this);
    this.onModalCancel = this.onModalCancel.bind(this);
    this.onFormatChange = this.onFormatChange.bind(this);
    this.onDownloadClick = this.onDownloadClick.bind(this);

    this.state = {
      modalVisible: false,
      errorMessage: null,
    };
    this.citeContentCacheByFormat = {};
  }

  onCiteClick() {
    const { citeContent } = this.state;
    this.setState({ modalVisible: true });

    // initial modal open
    if (!citeContent) {
      const { initialCiteFormat } = this.props;
      this.setCiteContentFor(initialCiteFormat);
    }
  }

  onDownloadClick() {
    const { citeContent } = this.state;
    downloadTextAsFile(citeContent);
  }

  onModalCancel() {
    this.setState({ modalVisible: false });
  }

  onFormatChange(format) {
    const { onCiteFormatChange } = this.props;
    onCiteFormatChange(format);
    this.setCiteContentFor(format);
  }

  async setCiteContentFor(format) {
    let citeContent = this.citeContentCacheByFormat[format];
    if (!citeContent) {
      const { recordId } = this.props;
      try {
        citeContent = await citeArticle(format, recordId);
        this.citeContentCacheByFormat[format] = citeContent;
        this.setState({ errorMessage: null });
      } catch (error) {
        this.setState({
          errorMessage: `Could not create cite text for the selected format. Caused by: ${
            error.message
          }`,
        });
      }
    }

    this.setState({ citeContent });
  }

  render() {
    const { initialCiteFormat } = this.props;
    const { modalVisible, citeContent, errorMessage } = this.state;
    return (
      <>
        <ListItemAction>
          <EventTracker eventId="Cite">
            <Button onClick={this.onCiteClick}>
              <IconText text="cite" type="export" />
            </Button>
          </EventTracker>
        </ListItemAction>
        <Modal
          title="Cite Article"
          visible={modalVisible}
          footer={null}
          onCancel={this.onModalCancel}
        >
          <div>
            {errorMessage && (
              <div className="mb3">
                <Alert type="error" showIcon description={errorMessage} />
              </div>
            )}
            <Row>
              <pre>{citeContent}</pre>
            </Row>
            <Row type="flex" justify="space-between">
              <div>
                <CopyToClipboard text={citeContent} onCopy={this.onModalCancel}>
                  <Button style={{ marginRight: 12 }}>
                    <Icon type="copy" /> Copy to Clipboard
                  </Button>
                </CopyToClipboard>
                <Button onClick={this.onDownloadClick}>
                  <Icon type="download" /> Download
                </Button>
              </div>
              <EventTracker
                eventId="CiteFormatSelection"
                eventPropName="onChange"
                extractEventArgsToForward={args => [args[0]]}
              >
                <SelectBox
                  style={{ width: 140 }}
                  defaultValue={initialCiteFormat}
                  onChange={this.onFormatChange}
                  options={CITE_FORMAT_OPTIONS}
                />
              </EventTracker>
            </Row>
          </div>
        </Modal>
      </>
    );
  }
}

CiteModalAction.propTypes = {
  recordId: PropTypes.number.isRequired,
  initialCiteFormat: PropTypes.oneOf(CITE_FORMAT_VALUES).isRequired,
  onCiteFormatChange: PropTypes.func.isRequired,
};

export default CiteModalAction;
