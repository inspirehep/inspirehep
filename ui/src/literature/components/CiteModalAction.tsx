import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  CopyOutlined,
  DownloadOutlined,
  ExportOutlined,
} from '@ant-design/icons';
import { Modal, Button, Row, Alert } from 'antd';
// @ts-expect-error ts-migrate(7016) FIXME: Could not find a declaration file for module 'reac... Remove this comment to see the full error message
import { CopyToClipboard } from 'react-copy-to-clipboard';

import SelectBox from '../../common/components/SelectBox';
import ListItemAction from '../../common/components/ListItemAction';
import IconText from '../../common/components/IconText';
import EventTracker from '../../common/components/EventTracker';
import {
  CITE_FORMAT_OPTIONS,
  CITE_FORMAT_VALUES,
  CITE_FILE_FORMAT,
  CV,
} from '../constants';
import citeArticle from '../citeArticle';
import { downloadTextAsFile } from '../../common/utils';
import RichDescription from '../../common/components/RichDescription';
import LoadingOrChildren from '../../common/components/LoadingOrChildren';

class CiteModalAction extends Component {
  
  constructor(props: any) {
    super(props);
    this.onCiteClick = this.onCiteClick.bind(this);
    this.onModalCancel = this.onModalCancel.bind(this);
    this.onFormatChange = this.onFormatChange.bind(this);
    this.onDownloadClick = this.onDownloadClick.bind(this);
    
    this.state = {
      modalVisible: false,
      errorMessage: null,
      loading: false,
    };
    this.citeContentCacheByFormat = {};
  }
  
  onCiteClick() {
    // @ts-expect-error ts-migrate(2339) FIXME: Property 'citeContent' does not exist on type 'Rea... Remove this comment to see the full error message
    const { citeContent } = this.state;
    this.setState({ modalVisible: true });
    
    // initial modal open
    if (!citeContent) {
      // @ts-expect-error ts-migrate(2339) FIXME: Property 'initialCiteFormat' does not exist on typ... Remove this comment to see the full error message
      const { initialCiteFormat } = this.props;
      this.setCiteContentFor(initialCiteFormat);
    }
  }
  
  onDownloadClick() {
    // @ts-expect-error ts-migrate(2339) FIXME: Property 'citeContent' does not exist on type 'Rea... Remove this comment to see the full error message
    const { citeContent, format } = this.state;
    // @ts-expect-error ts-migrate(2339) FIXME: Property 'recordId' does not exist on type 'Readon... Remove this comment to see the full error message
    const { recordId } = this.props;
    downloadTextAsFile(
      citeContent,
      // @ts-expect-error ts-migrate(7053) FIXME: Element implicitly has an 'any' type because expre... Remove this comment to see the full error message
      `INSPIRE-Cite-${recordId}.${CITE_FILE_FORMAT[format].extension}`,
      // @ts-expect-error ts-migrate(7053) FIXME: Element implicitly has an 'any' type because expre... Remove this comment to see the full error message
      CITE_FILE_FORMAT[format].mimetype
      );
    }
    
    onModalCancel() {
      this.setState({ modalVisible: false });
    }
    
    onFormatChange(format: any) {
      // @ts-expect-error ts-migrate(2339) FIXME: Property 'onCiteFormatChange' does not exist on ty... Remove this comment to see the full error message
      const { onCiteFormatChange } = this.props;
      onCiteFormatChange(format);
      this.setCiteContentFor(format);
    }
    
    async setCiteContentFor(format: any) {
      // TODO: remove this cache and rely on the browser http caching
      let citeContent = this.citeContentCacheByFormat[format];
      if (!citeContent) {
        // @ts-expect-error ts-migrate(2339) FIXME: Property 'recordId' does not exist on type 'Readon... Remove this comment to see the full error message
        const { recordId } = this.props;
        this.setState({ loading: true });
        try {
          citeContent = await citeArticle(format, recordId);
          this.citeContentCacheByFormat[format] = citeContent;
          this.setState({ errorMessage: null, loading: false });
        } catch (error) {
          this.setState({
            errorMessage: `Could not create cite text for the selected format. Caused by: ${
              // @ts-expect-error ts-migrate(2571) FIXME: Object is of type 'unknown'.
              error.message
            }`,
            loading: false,
          });
        }
      }
      
      this.setState({ citeContent, format });
    }
    
    citeContentCacheByFormat: any;
    
  render() {
    // @ts-expect-error ts-migrate(2339) FIXME: Property 'initialCiteFormat' does not exist on typ... Remove this comment to see the full error message
    const { initialCiteFormat } = this.props;
    const {
      // @ts-expect-error ts-migrate(2339) FIXME: Property 'modalVisible' does not exist on type 'Re... Remove this comment to see the full error message
      modalVisible,
      // @ts-expect-error ts-migrate(2339) FIXME: Property 'citeContent' does not exist on type 'Rea... Remove this comment to see the full error message
      citeContent,
      // @ts-expect-error ts-migrate(2339) FIXME: Property 'errorMessage' does not exist on type 'Re... Remove this comment to see the full error message
      errorMessage,
      // @ts-expect-error ts-migrate(2339) FIXME: Property 'format' does not exist on type 'Readonly... Remove this comment to see the full error message
      format,
      // @ts-expect-error ts-migrate(2339) FIXME: Property 'loading' does not exist on type 'Readonl... Remove this comment to see the full error message
      loading,
    } = this.state;

    const loadingOrChildrenProps = {
      loading
    }

    const iconTextProps = {
      text: "cite",
      icon: <ExportOutlined />
    }

    const eventTrackerProps = {
      eventId: "Cite"
    }

    return <>
      <ListItemAction>
        <EventTracker {...eventTrackerProps}>
          <Button onClick={this.onCiteClick}>
            <IconText {...iconTextProps} />
          </Button>
        </EventTracker>
      </ListItemAction>
      <Modal
        title="Cite Article"
        visible={modalVisible}
        footer={null}
        onCancel={this.onModalCancel}
      >
        <LoadingOrChildren {...loadingOrChildrenProps}>
          <div>
            {errorMessage && (
              <div className="mb3">
                <Alert type="error" showIcon description={errorMessage} message={false} />
              </div>
            )}
            <Row>
              {format === CV ? (
                <RichDescription>{citeContent}</RichDescription>
              ) : (
                <pre>{citeContent}</pre>
              )}
            </Row>
            <Row justify="space-between">
              <div>
                <CopyToClipboard
                  text={citeContent}
                  onCopy={this.onModalCancel}
                >
                  <Button style={{ marginRight: 12 }}>
                    <CopyOutlined /> Copy to Clipboard
                  </Button>
                </CopyToClipboard>
                <Button onClick={this.onDownloadClick}>
                  <DownloadOutlined /> Download
                </Button>
              </div>
              <EventTracker
                // @ts-expect-error ts-migrate(2322) FIXME: Type '{ children: Element; eventId: string; eventP... Remove this comment to see the full error message
                eventId="CiteFormatSelection"
                eventPropName="onChange"
                extractEventArgsToForward={(args: any) => [args[0]]}
              >
                <SelectBox
                  // @ts-expect-error ts-migrate(2322) FIXME: Type '{ style: { width: number; }; defaultValue: a... Remove this comment to see the full error message
                  style={{ width: 140 }}
                  defaultValue={initialCiteFormat}
                  onChange={this.onFormatChange}
                  options={CITE_FORMAT_OPTIONS}
                />
              </EventTracker>
            </Row>
          </div>
        </LoadingOrChildren>
      </Modal>
    </>;
  }
}

// @ts-expect-error ts-migrate(2339) FIXME: Property 'propTypes' does not exist on type 'typeo... Remove this comment to see the full error message
CiteModalAction.propTypes = {
  recordId: PropTypes.number.isRequired,
  initialCiteFormat: PropTypes.oneOf(CITE_FORMAT_VALUES).isRequired,
  onCiteFormatChange: PropTypes.func.isRequired,
};

export default CiteModalAction;
