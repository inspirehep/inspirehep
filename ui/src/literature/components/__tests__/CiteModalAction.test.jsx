import React from 'react';
import { shallow } from 'enzyme';
import { Modal, Button } from 'antd';

import CiteModalAction from '../CiteModalAction';
import citeArticle from '../../citeArticle';
import SelectBox from '../../../common/components/SelectBox';
import UserAction from '../../../common/components/UserAction';
import { CITE_FORMAT_VALUES } from '../../constants';

jest.mock('../../citeArticle');

describe('CiteModalAction', () => {
  beforeAll(() => {
    citeArticle.mockImplementation(
      (format, recordId) => `Cite ${recordId} in ${format}`
    );
  });

  it('renders with all props', () => {
    const wrapper = shallow(
      <CiteModalAction
        recordId={12345}
        initialCiteFormat="application/x-bibtex"
        onCiteFormatChange={jest.fn()}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('calls onCiteFormatChange on format change', () => {
    const onCiteFormatChangeProp = jest.fn();
    const wrapper = shallow(
      <CiteModalAction
        recordId={12345}
        initialCiteFormat={CITE_FORMAT_VALUES[0]}
        onCiteFormatChange={onCiteFormatChangeProp}
      />
    );
    wrapper.find(SelectBox).prop('onChange')(CITE_FORMAT_VALUES[1]);
    expect(onCiteFormatChangeProp).toHaveBeenCalledWith(CITE_FORMAT_VALUES[1]);
  });

  it('sets modalVisible true on cite click and calls setCiteContentFor with initialCiteFormat if first time', () => {
    const initialCiteFormat = 'application/x-bibtex';
    const wrapper = shallow(
      <CiteModalAction
        recordId={12345}
        initialCiteFormat={initialCiteFormat}
        onCiteFormatChange={jest.fn()}
      />
    );
    const setCiteContentFor = jest.fn();
    wrapper.instance().setCiteContentFor = setCiteContentFor;
    wrapper.update();
    const onCiteClick = wrapper.find(UserAction).find(Button).prop('onClick');
    onCiteClick();
    expect(wrapper.state('modalVisible')).toBe(true);
    expect(setCiteContentFor).toHaveBeenCalledWith(initialCiteFormat);
  });

  it('sets modalVisible true on cite click but does not call setCiteContentFor if citeContent is present', () => {
    const wrapper = shallow(
      <CiteModalAction
        recordId={12345}
        initialCiteFormat="application/x-bibtex"
        onCiteFormatChange={jest.fn()}
      />
    );
    const setCiteContentFor = jest.fn();
    wrapper.instance().setCiteContentFor = setCiteContentFor;
    wrapper.setState({ citeContent: 'CONTENT' });
    wrapper.update();
    const onCiteClick = wrapper.find(UserAction).find(Button).prop('onClick');
    onCiteClick();
    expect(wrapper.state('modalVisible')).toBe(true);
    expect(setCiteContentFor).not.toHaveBeenCalled();
  });

  it('shows an alert with the error message when there is an error in setCiteContentFor', () => {
    const initialCiteFormat = 'application/x-bibtex';
    const wrapper = shallow(
      <CiteModalAction
        recordId={12345}
        initialCiteFormat={initialCiteFormat}
        onCiteFormatChange={jest.fn()}
      />
    );
    const errorMessage = 'There is an error';
    wrapper.setState({ errorMessage });
    expect(wrapper).toMatchSnapshot();
  });

  it('sets citeContent for selected format setCiteContentFor', async () => {
    const wrapper = shallow(
      <CiteModalAction
        recordId={12345}
        initialCiteFormat="application/x-bibtex"
        onCiteFormatChange={jest.fn()}
      />
    );
    const setCiteContentFor = wrapper.find(SelectBox).prop('onChange');
    await setCiteContentFor('application/vnd+inspire.latex.us+x-latex');
    expect(wrapper.state('citeContent')).toEqual(
      'Cite 12345 in application/vnd+inspire.latex.us+x-latex'
    );
  });

  it('sets modalVisible false onModalCancel', () => {
    const wrapper = shallow(
      <CiteModalAction
        recordId={12345}
        initialCiteFormat="application/x-bibtex"
        onCiteFormatChange={jest.fn()}
      />
    );
    wrapper.instance().state.modalVisible = true;
    wrapper.update();
    const onModalCancel = wrapper.find(Modal).prop('onCancel');
    onModalCancel();
    expect(wrapper.state('modalVisible')).toBe(false);
  });

  it('renders with loading', () => {
    const initialCiteFormat = 'application/x-bibtex';
    const wrapper = shallow(
      <CiteModalAction
        recordId={12345}
        initialCiteFormat={initialCiteFormat}
        onCiteFormatChange={jest.fn()}
      />
    );
    wrapper.setState({ loading: true });
    expect(wrapper).toMatchSnapshot();
  });
});
