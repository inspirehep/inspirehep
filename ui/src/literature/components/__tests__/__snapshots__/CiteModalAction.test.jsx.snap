// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CiteModalAction renders with all props 1`] = `
<Fragment>
  <ListItemAction>
    <EventTracker
      eventId="Cite"
      eventPropName="onClick"
      extractEventArgsToForward={null}
    >
      <Button
        block={false}
        ghost={false}
        htmlType="button"
        loading={false}
        onClick={[Function]}
      >
        <IconText
          text="cite"
          type="export"
        />
      </Button>
    </EventTracker>
  </ListItemAction>
  <Modal
    confirmLoading={false}
    footer={null}
    maskTransitionName="fade"
    okType="primary"
    onCancel={[Function]}
    title="Cite Article"
    transitionName="zoom"
    visible={false}
    width={520}
  >
    <div>
      <Row
        gutter={0}
      >
        <pre />
      </Row>
      <Row
        gutter={0}
        justify="space-between"
        type="flex"
      >
        <div>
          <CopyToClipboard
            onCopy={[Function]}
          >
            <Button
              block={false}
              ghost={false}
              htmlType="button"
              loading={false}
              style={
                Object {
                  "marginRight": 12,
                }
              }
            >
              <Icon
                type="copy"
              />
               Copy to Clipboard
            </Button>
          </CopyToClipboard>
          <Button
            block={false}
            ghost={false}
            htmlType="button"
            loading={false}
            onClick={[Function]}
          >
            <Icon
              type="download"
            />
             Download
          </Button>
        </div>
        <EventTracker
          eventId="CiteFormatSelection"
          eventPropName="onChange"
          extractEventArgsToForward={[Function]}
        >
          <SelectBox
            defaultValue="x-bibtex"
            onChange={[Function]}
            options={
              Array [
                Object {
                  "display": "BibTeX",
                  "value": "x-bibtex",
                },
                Object {
                  "display": "LaTeX (EU)",
                  "value": "vnd+inspire.latex.eu+x-latex",
                },
                Object {
                  "display": "LaTeX (US)",
                  "value": "vnd+inspire.latex.us+x-latex",
                },
              ]
            }
            style={
              Object {
                "width": 140,
              }
            }
          />
        </EventTracker>
      </Row>
    </div>
  </Modal>
</Fragment>
`;

exports[`CiteModalAction shows an alert with the error message when there is an error in setCiteContentFor 1`] = `
<Fragment>
  <ListItemAction>
    <EventTracker
      eventId="Cite"
      eventPropName="onClick"
      extractEventArgsToForward={null}
    >
      <Button
        block={false}
        ghost={false}
        htmlType="button"
        loading={false}
        onClick={[Function]}
      >
        <IconText
          text="cite"
          type="export"
        />
      </Button>
    </EventTracker>
  </ListItemAction>
  <Modal
    confirmLoading={false}
    footer={null}
    maskTransitionName="fade"
    okType="primary"
    onCancel={[Function]}
    title="Cite Article"
    transitionName="zoom"
    visible={false}
    width={520}
  >
    <div>
      <div
        className="mb3"
      >
        <Alert
          description="There is an error"
          showIcon={true}
          type="error"
        />
      </div>
      <Row
        gutter={0}
      >
        <pre />
      </Row>
      <Row
        gutter={0}
        justify="space-between"
        type="flex"
      >
        <div>
          <CopyToClipboard
            onCopy={[Function]}
          >
            <Button
              block={false}
              ghost={false}
              htmlType="button"
              loading={false}
              style={
                Object {
                  "marginRight": 12,
                }
              }
            >
              <Icon
                type="copy"
              />
               Copy to Clipboard
            </Button>
          </CopyToClipboard>
          <Button
            block={false}
            ghost={false}
            htmlType="button"
            loading={false}
            onClick={[Function]}
          >
            <Icon
              type="download"
            />
             Download
          </Button>
        </div>
        <EventTracker
          eventId="CiteFormatSelection"
          eventPropName="onChange"
          extractEventArgsToForward={[Function]}
        >
          <SelectBox
            defaultValue="x-bibtex"
            onChange={[Function]}
            options={
              Array [
                Object {
                  "display": "BibTeX",
                  "value": "x-bibtex",
                },
                Object {
                  "display": "LaTeX (EU)",
                  "value": "vnd+inspire.latex.eu+x-latex",
                },
                Object {
                  "display": "LaTeX (US)",
                  "value": "vnd+inspire.latex.us+x-latex",
                },
              ]
            }
            style={
              Object {
                "width": 140,
              }
            }
          />
        </EventTracker>
      </Row>
    </div>
  </Modal>
</Fragment>
`;
